Implement integredient groups for VarechaPravdaSk.

jose1711 opened the pull request.

jknndy commented:
Hi @jose1711 , thanks for the contribution!

First, for ingredient group coverage we require a separate test cases that covers a page that contains the groupings. Secondly, it appears that this is causing some issues with the ingredients vs ingredient_groups test. After looking into it myself it appears that there may be a simplier way to approach gathering the grouped ingredients. I was able to get a full list of the groupings and also match the output of ingredients using the code below. Let me know what you think!
```python
from ._grouping_utils import group_ingredients

def ingredient_groups(self):
    return group_ingredients(
        self.ingredients(),
        self.soup,
        ".recipe-ingredients__group",
        ".recipe-ingredients__row",
    )
```

jose1711 commented:
thank you @jknndy 
your solution is much tidier. i updated the class slightly to use a colon delimiter between ingredient name and amount. also added test.

jknndy reviewed (changes requested):
A few more small things then we should be good to go!

jose1711 commented on recipe_scrapers/varechapravdask.py:
```diff
@@ -1,7 +1,37 @@
+        )
+
+        if ingredient_amount:
+            return f"{ingredient_name}: {ingredient_amount}"
```
@jknndy The format of ingredients and amount display is actually a bit unclear to me. There is no standardized approach? Or we are simply mirroring the format recipe site is using?

jknndy replied:
You're right there is no standard implementation but we try to represent the recipe data as close to the source as possible. Also traditionally (at least in English) it would be quantity, unit and food for the order although some sites do display it otherwise.

jose1711 replied:
Thank you, updated accordingly.

jose1711 commented on recipe_scrapers/varechapravdask.py:
```diff
@@ -1,7 +1,37 @@
+
+    def _format_ingredient(self, element):
+        ingredient_name = normalize_string(
+            element.find(attrs={"recipe-ingredients__ingredient"}).text
```
Thank you, updated.

jose1711 commented on tests/test_data/varecha.pravda.sk/varechapravdask_2.json:
```diff
@@ -0,0 +1,91 @@
+  ],
+  "yields": "15 servings",
+  "description": "Nech si nemusíme posúvať recepty na sladko a na slano, pridávam aj na slano, ale to musím napísať, ako som sa inšpirovala. Inšpirovala som sa receptom na syrové rožteky od @Maxinka, lebo bez jej receptu by ma to ani &quot;netrklo&quot; tvarovať rožteky. Ja som si prispôsobila zloženie cesta a tvar výsledneho pečiva podľa svojej chuti a chuti našej rodiny. Určite nekopírujem, rec ...
+  "total_time": 110,
```
Thank you, updated.

hhursev commented:
The final version is very clean. GJ to both of you!
